src/java/org/apache/cassandra/index/sai/disk/v1/PerIndexFiles.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v1;
+
+import java.io.Closeable;
+import java.util.EnumMap;
+import java.util.Map;
+
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.disk.format.IndexComponent;
+import org.apache.cassandra.index.sai.disk.format.IndexDescriptor;
+import org.apache.cassandra.index.sai.utils.TypeUtil;
+import org.apache.cassandra.io.util.FileHandle;
+import org.apache.cassandra.io.util.FileUtils;
+
+public class PerIndexFiles implements Closeable
```

Comment:
```suggestion
public class PerColumnIndexFiles implements Closeable
```
?

Reply:
Looking at this class. All usages of `getFile` make a `sharedCopy` of the `FileHandle`. It would make sense if we pull that functionality into this class so that we can't inadvertently not make a copy.

WDYT?

Reply:
Yeah, if we don't imagine any usage where we actually want a new `Ref`, that makes sense. Perhaps just a quick blurb in the JavaDoc for the private `getFile()` that mentions it, etc.